core/src/main/java/hudson/lifecycle/UnixLifecycle.java

```diff
@@ -84,10 +90,14 @@ public void restart() throws IOException, InterruptedException {
             LIBC.fcntl(i, F_SETFD,flags| FD_CLOEXEC);
         }
 
-        // exec to self
-        String exe = args.get(0);
-        LIBC.execvp(exe, new StringArray(args.toArray(new String[args.size()])));
-        throw new IOException("Failed to exec '"+exe+"' "+LIBC.strerror(Native.getLastError()));
+        if (EXIT_ON_RESTART == null) {
+            // exec to self
+            String exe = args.get(0);
+            LIBC.execvp(exe, new StringArray(args.toArray(new String[args.size()])));
+            throw new IOException("Failed to exec '"+exe+"' "+LIBC.strerror(Native.getLastError()));
+        } else {
+            System.exit(Integer.parseInt(EXIT_ON_RESTART));
```

Comment:
The field should be `Integer` then. No need to parse it in runtime and to risk that the logic blows up here in the case of typo in the argument

Reply:
ok